breakpad-symbols/src/sym_file/types.rs

```diff
@@ -58,6 +79,56 @@ impl Function {
             self.address.checked_add(self.size as u64)? - 1,
         ))
     }
+
+    /// Returns `(file_id, line, address)` of the line or inline record that
+    /// covers the given address at the outermost level (i.e. not inside any
+    /// inlined calls).
+    pub fn get_outermost_source_location(&self, addr: u64) -> Option<(u32, u32, u64)> {
+        if let Some((call_file, call_line, address, _)) = self.get_inlinee_at_depth(0, addr) {
+            return Some((call_file, call_line, address));
+        }
+        let line = self.lines.get(addr)?;
+        Some((line.file, line.line, line.address))
+    }
+
+    /// Returns `(file_id, line, address)` of the line record that covers the
+    /// given address. Line records describe locations at the innermost level
+    /// of inline calls at that address.
+    pub fn get_innermost_source_location(&self, addr: u64) -> Option<(u32, u32, u64)> {
+        let line = self.lines.get(addr)?;
+        Some((line.file, line.line, line.address))
+    }
```

Comment:
Is there a specific purpose for finding the innermost?

Reply:
Not at the moment, but there will be once we build the inline stack. Usually the way it works is that you build the inline stack from "outside to inside" by getting inlines at depth k for k in 0.. until there is no inline at that depth, and then you get the "innermost source location" for the final frame's location.
We now have a public method for getting the inline, so it felt natural to add another public method to get the innermost line as well. But I can leave it out until we have a use for it.

Reply:
No it's fine if it has an eventual purpose. Is the key point here that `get_inlinee_at_depth` can never actually yield the value of `innermost_source_location`, so you need to carve that our a final step?

Reply:
That's right.